src/Konami.tsx

```diff
@@ -0,0 +1,167 @@
+import React from 'react';
+
+export interface KonamiProps {
+  className?: string;
+  code?: number[];
+  disabled?: boolean;
+  resetDelay?: number;
+  action?: () => void;
+  timeout?: number;
+  onTimeout?: () => void;
+}
+
+export interface KonamiStates {
+  done: boolean;
+  input: number[];
+}
+
+class Timer {
+  t: number;
+  timerObj: number | null;
+  fn: TimerHandler;
+
+  constructor(fn: () => void, t: number) {
+    this.t = t;
+    this.fn = fn;
+
+    this.timerObj = setInterval(this.fn, this.t);
+  }
+
+  stop() {
+    if (this.timerObj) {
+      clearInterval(this.timerObj);
+      this.timerObj = null;
+    }
+    return this;
+  }
+
+  start() {
+    if (!this.timerObj) {
+      this.stop();
+      this.timerObj = setInterval(this.fn, this.t);
+    }
+    return this;
+  }
+
+  reset(newT: number) {
+    this.t = newT;
+    return this.stop().start();
+  }
+}
+
+class Konami extends React.Component<KonamiProps, KonamiStates> {
+  private timeoutFunc: ReturnType<typeof setTimeout> | null | undefined;
+  private _timer: any;
+
+  static defaultProps = {
+    action: null,
+    className: '',
+    code: [38, 38, 40, 40, 37, 39, 37, 39, 66, 65],
+    disabled: false,
+    onTimeout: null,
+    resetDelay: 1000,
+    timeout: null,
+  };
+
+  constructor(props: KonamiProps) {
+    super(props);
+
+    this.state = {
+      done: false,
+      input: [],
+    };
+
+    this.timeoutFunc = null;
+    this.onKeyUp = this.onKeyUp.bind(this);
+    this.resetInput = this.resetInput.bind(this);
+  }
+
+  componentDidMount() {
+    const { resetDelay } = this.props;
+
+    document.addEventListener('keyup', this.onKeyUp);
+    const delay = Number(resetDelay);
+    if (delay !== 0) {
+      this._timer = new Timer(() => this.resetInput(), delay);
+    }
+  }
+
+  componentWillUnmount() {
+    const { resetDelay } = this.props;
+
+    clearTimeout(Number(this.timeoutFunc ?? undefined));
+    if (resetDelay !== 0) {
+      this._timer.stop();
+    }
+    document.removeEventListener('keyup', this.onKeyUp);
+  }
+
+  onKeyUp(e: KeyboardEvent) {
+    const { done, input } = this.state;
+    const {
+      action,
+      code,
+      disabled,
+      onTimeout,
+      resetDelay,
+      timeout,
+    } = this.props;
+
+    const delay = Number(resetDelay);
+
+    if (disabled) {
+      return;
+    }
+
+    if (delay !== 0) {
+      this._timer.reset(delay);
+    }
+
+    input.push(e.keyCode);
+
+    if (code) input.splice(-code.length - 1, input.length - code.length);
+
+    this.setState({ input }, () => {
+      if (this.state.input.join('').includes(code?.join('') ?? '') && !done) {
+        /* eslint-disable-line */
```

Comment:
Shouldn't the `/* eslint-disable-line */` be in the same line as the code you're trying to exempt? Otherwise put it above the line and use `// eslint-disable-next-line`. Did the ESLint error disappear with the change to TS?

Reply:
Yes, I haven't received an ES error. So I will remove that line, can I?

Reply:
If there's no ESLint warning/error anymore it should be removed, yes.